### ISSUE TITLE
ideas for a v2.0 release

### ISSUE THREAD

**Issue by millermedeiros** [millermedeiros/crossroads.js#82]
Some ideas/improvements for a future _not backwards compatible_ release:
## no global/static router

So instead of having access to a global `crossroads` router you would need to
create your own instance. I think this would encourage people to create their
own instances and not rely on a global object.

``` js
// instead of
crossroads.addRoute(...);
// you would need to create a new instance
var myRouter = crossroads.create();
myRouter.addRoute(...);
```

see: #55 and #19
## remove "greedy" routes support

As discussed before I think this feature is unnecessary when you have option
to _pipe_ multiple routers and I think it was a very bad design decision.

see: #68
## Route ID + no listener on `addRoute()`

Some people asked for a way to retrieve a route by their ID so they could use
`Router#interpolate()` more easily through the app.

So far I haven't needed this feature (since I usually code something similar
inside my app structure) but it might make things simpler to some
users. Thinking in something like:

``` js
// notice that there is no way to set a listener directly you need to
// use the "matched" signal.
var newsRoute = myRouter.addRoute('news', 'news/{id}', priority);
newsRoute.matched.add(doSomething);
// later you can retrieve the route
var path = myRouter.getRoute('news').interpolate({id:123});
// and also simplify removal
myRouter.removeRoute('news');
```

I think it will make it clearer that the signal can have multiple listeners.

Maybe ID can be optional, so if user doesn't provide it the router uses the
pattern as the ID instead:

``` js
router.addRoute('foo/{bar}');
router.getRoute('foo/{bar}').interpolate({id:123});
```

see: #38
## Remove typecasting or make it a local setting

There are many cases where it can cause conflicts since you don't expect it to
happen on a few cases. I think all values should be treated as strings and user
can typecast the value inside his own app.

See: #81 and [mout/string/typecast](http://moutjs.com/docs/latest/string.html#typecast)
## Add `route.trigger()`

Basically the same as calling `myRouter.parse( route.interpolate({id:123}) )`
just as a convenince.

**Comment by killmenot** [comment]
@millermedeiros the ideas you described above, did you discard them? To be honest I'm very interesting in the correct filters support. Now, I use `matched/switched` for that but they have no any `cancel()` logic

**Comment by millermedeiros** [comment]
@killmenot I never got the time to work on this, I started working inside bigger projects and mostly maintaining legacy apps.. the last time I used crossroads on any project was back in 2013, so that's why the development/support here been stale ever since. - I was mostly developing new features and fixing bugs as needed...
